### DISCUSSION TITLE
Issues mapping Enum "typeMapping"

### DISCUSSION THREAD

**Discussion by evanmalmud** [Netflix/dgs-framework#1358] [General]
So Im trying to make the dgs code gen (maven community version) typeMapping from a enum in graphql to a java enum.
Im having issues getting this to map and startup correctly. Once i add @DgsScalar to the enum i get an error about missing a String java bean. Which makes me thing its not possible to map an enum this way.
```
Parameter 0 of constructor in com.Status a bean of type 'java.lang.String' that could not be found.
```

Was trying to add the coercing implementation to the java type as well but without DGSScalar i figure this wont help.

My java enum - 

```
@DgsScalar(name = "GQLStatus")
public enum Status {

  FINISHED,
  PENDING,
  FAILED,
  REQUESTED,
  UNREQUESTED;
```
Graphql Enum - 
```
"""
Response status enum on form submit
"""
enum GQLStatus {
    FINISHED
    PENDING
    FAILED
    REQUESTED
    UNREQUESTED
}
```

**Comment by evanmalmud** [comment]
So without the DGSScalar or any Coercing the mapping is generated into the parent codegen classes and all seems to functions correctly. 
But the codegen still create the enum class with the original name which made me think initially it wasnt being picked up.

**Comment by deweyjose** [comment]
Just following up here - there may have been some initial confusion. There isn't an issue with the maven community version of the codegen plugin. 

Adding an entry for the GraphQL => Java enum in `typeMapping` works as expected with a similar example for me.
